### ISSUE TITLE
v:page.resources.fal returns empty array when no limit is specified

### ISSUE THREAD

**Issue by binaryfs** [FluidTYPO3/vhs#1828]
Hello,

as already mentioned in the title, the v:page.resources.fal ViewHelper returns an empty array when no limit is specified:
```xml
<v:page.resources.fal table="pages" field="media" uid="{someUid}" as="images">
    <!-- {images} is an empty array -->
</v:page.resources.fal>
```

With limit everything works fine:
```xml
<v:page.resources.fal table="pages" field="media" uid="{someUid}" as="images" limit="1">
```

This is caused by the `SlideViewHelperTrait`. If no limit is given, `$limit` will be null in this method:

```php
protected function getSlideRecords($pageUid, $limit = null)
{
    if (null === $limit && false === empty($this->arguments['limit'])) {
        $limit = (integer) $this->arguments['limit'];
    }
```

And thus the following condition evaluates to true, even though it shouldn't:
```php
if (0 !== $limit) {
```

This behavior was introduced in 6.1.3. My current workaround is to always set a limit in the ViewHelper.

Best regards

**Comment by olegkarun** [comment]
Same for me

**Comment by Oktopuce** [comment]
Same problem for me with TYPO3 V11 and PHP 8.

**Comment by fischhase** [comment]
same with V11 and PHP 7.4

**Comment by ursbraem** [comment]
Same with TYPO3 10.4.34 and PHP 7.4

**Comment by NamelessCoder** [comment]
Fixed since 9b64157959d6237efd8d4d8c0aeb510e10b9fc36 and 3b4c8e04649ad9de523b4413b1fab0ca24d53a22.
